cpp/include/cudf/detail/utilities/cuda_memcpy.hpp

```diff
@@ -71,6 +71,26 @@ void cuda_memcpy_async(host_span<T> dst, device_span<T const> src, rmm::cuda_str
                          stream);
 }
 
+/**
+ * @brief Asynchronously copies data from host memory to host memory.
+ *
+ * Implementation may use different strategies depending on the size and type of host data.
+ *
+ * @param dst Destination host memory
+ * @param src Source device memory
+ * @param stream CUDA stream used for the copy
+ */
+template <typename T>
+void cuda_memcpy_async(host_span<T> dst, host_span<T const> src, rmm::cuda_stream_view stream)
```

Comment:
Using a `cuda_memcpy_async` named utility for host-to-host copies seems super fishy to me 😄